packages/playground/windows/playground-win32/Playground-Win32.cpp

```diff
@@ -0,0 +1,447 @@
+              std::make_shared<PlaygroundNativeModuleProvider>();
+          std::shared_ptr<react::uwp::ViewManagerProvider> viewManagerProvider =
+              std::make_shared<PlaygroundViewManagerProvider>();
+
+          m_instance = react::uwp::CreateReactInstance(moduleLoader, viewManagerProvider);
+
+          react::uwp::ReactInstanceSettings settings;
+          settings.UseWebDebugger = m_useWebDebugger;
+          settings.UseLiveReload = m_liveReloadEnabled;
+          settings.EnableDeveloperMenu = true;
+          settings.AsyncLocalStorageDBPath = GetAsyncLocalStorageDBPath();
+
+          settings.LoggingCallback = [](facebook::react::RCTLogLevel logLevel, const char *message) {
+            OutputDebugStringA("In LoggingCallback");
+            OutputDebugStringA(message);
+          };
+
+          settings.JsExceptionCallback = [](facebook::react::JSExceptionInfo &&exceptionInfo) {
+            OutputDebugStringA("in JsExceptionCallback");
+          };
+
+          m_instance->Start(m_instance, settings);
+          m_instance->loadBundle(Microsoft::Common::Unicode::Utf16ToUtf8(m_bundleFile));
+
+          folly::dynamic initialProps = folly::dynamic::object();
+
+          // Retrieve ABI pointer from C++/CX pointer
+          auto rootElement = m_desktopWindowXamlSource.Content().as<WUX::FrameworkElement>();
+
+          PCWSTR appName = (m_bundleFile == LR"(Samples\rntester)") ? L"RNTesterApp" : L"Bootstrap";
+
+          // Create the root view
+          m_rootView = react::uwp::CreateReactRootView(rootElement, appName, m_instanceCreator);
+
+          m_rootView->SetInitialProps(std::move(initialProps));
+          m_rootView->SetInstanceCreator(m_instanceCreator);
+          m_rootView->AttachRoot();
+        }
+
+        break;
+      }
+      case IDM_ABOUT:
+        DialogBox(s_instance, MAKEINTRESOURCE(IDD_ABOUTBOX), hwnd, &About);
+        break;
+      case IDM_EXIT:
+        PostQuitMessage(0);
+        break;
+      case IDM_REFRESH:
+        break;
+      case IDM_SETTINGS:
+        DialogBoxParam(s_instance, MAKEINTRESOURCE(IDD_SETTINGSBOX), hwnd, &Settings, reinterpret_cast<INT_PTR>(this));
+    }
+
+    return 0;
+  }
+
+  LRESULT OnCreate(HWND hwnd, LPCREATESTRUCT createStruct) {
+    auto interop = m_desktopWindowXamlSource.as<IDesktopWindowXamlSourceNative>();
+    // Parent the DesktopWindowXamlSource object to current window
+    winrt::check_hresult(interop->AttachToWindow(hwnd));
+
+    // Get the new child window's hwnd
+    HWND hWndXamlIsland = nullptr;
+    winrt::check_hresult(interop->get_WindowHandle(&hWndXamlIsland));
+
+    SetWindowPos(hWndXamlIsland, nullptr, 0, 0, createStruct->cx, createStruct->cy, SWP_SHOWWINDOW);
+
+    try {
+      m_instanceCreator = std::make_shared<HwndReactInstanceCreator>(hwnd);
+    } catch (...) {
+      return -1;
+    }
+
+    return 0;
+  }
+
+  LRESULT OnDestroy() {
+    PostQuitMessage(0);
```

Comment:
It doesn't matter that much since we're just going to kill the app anyway, but we should probably also run the WindowData destructor in WM_DESTROY if this isn't going to do cleanup.

Reply:
Oh nevermind, I see now that you're not intending for the window to have ownership of its data.

Reply:
Yeah, I agree with you here, should be free-ing data in `WM_DESTROY`.